Tests/RouterTests.swift

```diff
@@ -77,7 +88,95 @@ class RouterTests: QuickSpec {
             RouterTestServer.disable()
             Router.disableAll()
         }
-        
+
+        describe("Describe Cancelling requests") {
+            var router: Router!
+            let baseURL = "http://www.funky-cancel-request.com"
+            let latency = NSTimeInterval(2)
+
+            beforeEach {
+                router = Router.register(baseURL)
+                router.latency = latency // very high latency to allow us to stop the request before execution
+            }
+
+            it("should mark a request as cancelled") {
+                var responseError: NSError? = nil
+
+                router.get("/foobar/:id") { request in
+                    XCTFail("Request should get cancelled before execution")
+                    return nil
+                }
+
+                let requestURL = NSURL(string: "\(baseURL)/foobar/1")!
+
+                let dataTask = NSURLSession.sharedSession().dataTaskWithURL(requestURL) { (data, response, error) in
+                    responseError = error
+                }
+
+                dataTask.cancel()
+
+                expect(responseError).toEventually(beTruthy(), timeout: (latency + 1))
+                expect(responseError?.localizedDescription).toEventually(equal("cancelled"), timeout: (latency + 1))
+            }
+
+            it("should not confused multiple request with identical URL") {
+                var responseError_A: NSError? = nil
+                var responseError_B: NSError? = nil
+                let canceledRequestID = "999"
+
+                router.get("/cash/:id") { request in
+                    let paramID = request.components["id"]
+                    if paramID == canceledRequestID {
+                        XCTFail("Cancelled request should not get executed")
+                    }
+                    return nil
+                }
+
+                let requestURL_A = NSURL(string: "\(baseURL)/cash/333")!
+                let requestURL_B = NSURL(string: "\(baseURL)/cash/\(canceledRequestID)")!
+
+                let dataTask_A = NSURLSession.sharedSession().dataTaskWithURL(requestURL_A) { (data, response, error) in
+                    responseError_A = error
+                }
+                let dataTask_B = NSURLSession.sharedSession().dataTaskWithURL(requestURL_B) { (data, response, error) in
+                    responseError_B = error
+                }
+                dataTask_A.resume()
+                dataTask_B.cancel()
+
+                // expect task A to succeed
+                expect(responseError_A).toEventually(beNil(), timeout: (latency + 1))
```

Comment:
I would use a boolean instead of error for `A`:
e.g. requestSucceeded = false

`expect(requestSucceeded).toEventually(beTrue())`